### PR TITLE
docs: validator setup

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

- defined what variables needed to be changed in the `config.toml` in order to run a validator.
- Briefly explained how a sentry node archtecture should look

- [x] rebase prior to merging
______

For contributor use:

- ~~[ ] Wrote tests~~
- ~~[ ] Updated CHANGELOG_PENDING.md~~
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Updated relevant documentation (`docs/`) and code comments
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: docs/tendermint-core/validators.md [hunk 1]
```diff
@@ -17,14 +17,73 @@ to post any collateral at all.
...
+![ALT Local Configuration](./local_config.png)
+
+The validator will only talk to the sentry that are provided, the sentry nodes will communicate to the validator via a secret connection and the rest of the network through a normal connection. The sentry nodes do have the option of communicating with each other as well.
+
+When initializing nodes there are five parameters in the `config.toml` that may need to be altered.
+
+- `pex:` boolean. This turns the peer exchange reactor on or off for a node. When `pex=false`, only the `persistent_peers` list is available for connection.
+- `persistent_peers:` a comma separated list of `nodeID@ip:port` values that define a list of peers that are expected to be online at all times. This is necessary at first startup because by setting `pex=false` the node will not be able to join the network.
+- `unconditional_peer_ids:` comma separated list of nodeID's. These nodes will be connected to no matter the limits of inbound and outbound peers. This is useful for when sentry nodes have full address books.
+- `private_peer_ids:` comma separated list of nodeID's. These nodes will not be gossiped to the network. This is only necessary if `pex=false`.
```
Comment (cmwaters) [inline docs/tendermint-core/validators.md]: Isn't is when pex = true?

Comment (marbar3778) [reply docs/tendermint-core/validators.md]: reworded as it didn't make sense

File: docs/tendermint-core/validators.md [hunk 2]
```diff
@@ -17,14 +17,73 @@ to post any collateral at all.
 
 Validators have a cryptographic key-pair and an associated amount of
 "voting power". Voting power need not be the same.
+TestDuplicateVoteEvidenceValidation
```
Comment (melekes) [inline docs/tendermint-core/validators.md]: ???

File: docs/tendermint-core/validators.md [hunk 3]
```diff
@@ -17,14 +17,73 @@ to post any collateral at all.
...
+
+All sentry nodes (the “frontend”) connect to the validator using this private connection. The validator does not have a public IP address to provide its services.
+
+Amazon has multiple availability zones within a region. One can install sentry nodes in other regions too. In this case the second, third and further regions need to have a private connection to the validator node. This can be achieved by VPC Peering (“VPC Network Peering” in Google Cloud). In this case, the second, third and further region sentry nodes will be directed to the first region and through the direct connect to the data center, arriving to the validator.
+
+A more persistent solution (not detailed on the diagram) is to have multiple direct connections to different regions from the data center. This way VPC Peering is not mandatory, although still beneficial for the sentry nodes. This overcomes the risk of depending on one region. It is more costly.
```
Comment (melekes) [inline docs/tendermint-core/validators.md]: Should we also say non-cloud setup is also possible (i.e. bare-metal)? ^ implies that cloud is the only option

File: docs/tendermint-core/validators.md [hunk 4]
```diff
@@ -17,14 +17,73 @@ to post any collateral at all.
...
+| ---------------------- | --------------------------------------------- |
+| pex                    | true                                          |
+| persistent_peers       | validator node, optionally other sentry nodes |
+| private_peer_ids       | validator node ID                             |
+| unconditional_peer_ids | validator node ID, optionally sentry node IDs |
+| addr_book_strict       | false                                         |
+
+The sentry nodes should be able to talk to the entire network hence why `pex=true`. The persistent peers of a sentry node will be the validator, and optionally other sentry nodes. The sentry nodes should make sure that they do not gossip the validator's ip, to do this you must put the validators nodeID as a private peer. The unconditional peer IDs will be the validator ID and optionally other sentry nodes.
+
+> Note: Do not forget to secure your node's firewalls when setting them up.
```
Comment (melekes) [inline docs/tendermint-core/validators.md]: Should we link other useful guides here?

https://kb.certus.one/
https://forum.cosmos.network/t/sentry-node-architecture-overview/454

File: docs/tendermint-core/validators.md (hunk 3 above)
Comment (marbar3778) [reply docs/tendermint-core/validators.md]: added to the first paragraph in this section.

File: docs/tendermint-core/validators.md [hunk 5]
```diff
@@ -22,9 +22,72 @@ Validators have a cryptographic key-pair and an associated amount of
...
+
+### Network Layout
+
+![ALT Network Layout](./sentry_layout.png)
+
+The diagram is based on AWS, other cloud providers will have similar solutions to design a solution. Running nodes is not limited to cloud providers, you can run nodes on bare metal systems as well. The architecture will be the same no matter which setup you decide to go with.
+
+The proposed network diagram is similar to the classical backend/frontend separation of services in a corporate environment. The “backend” in this case is the private network of the validator in the data center. The data center network might involve multiple subnets, firewalls and redundancy devices, which is not detailed on this diagram. The important point is that the data center allows direct connectivity to the chosen cloud environment. Amazon AWS has “Direct Connect”, while Google Cloud has “Partner Interconnect”. This is a dedicated connection to the cloud provider (usually directly to your virtual private cloud instance in one of the regions).
```
Comment (erikgrinaker) [inline docs/tendermint-core/validators.md]: Do we generally not recommend running validators in cloud environments (even private ones), only in private data centers?

Comment (marbar3778) [reply docs/tendermint-core/validators.md]: People have recommended not to do this, but quite a few do anyways. I tried to leave it generalized, but may need a bit more work.

**Pull request merged**